here-naksha-lib-hub/src/main/java/com/here/naksha/lib/hub/mock/NHAdminWriterMock.java

```diff
@@ -104,14 +104,22 @@ public NHAdminWriterMock(final @NotNull Map<String, Map<String, Object>> mockCol
     // Perform write operation for each feature
     for (final XyzFeatureCodec featureCodec : wf.features) {
       // generate new feature Id, if not already provided
-      final XyzFeature feature = Objects.requireNonNull(featureCodec.getFeature(), "Codec's feature is null");
       final EWriteOp op = EWriteOp.get(featureCodec.getOp());
-      if (op == CREATE && (feature.getId() == null || feature.getId().isEmpty())) {
-        feature.setId(UUID.randomUUID().toString());
-      }
-      // persist feature in a space
       try {
-        XyzFeatureCodec result = null;
+        if ((op == DELETE) && (featureCodec.getFeature() == null)) {
+          if (featureCodec.getId() == null) {
```

Comment:
You may need to revisit this logic of raising exception. Attempting to delete a missing record, is not really a failure. It should be silently absorbed and should be ensured that the return list of features is having only those features which were actually found and deleted (excluding the missing ones).
Let consumer take a call, whether it should then return 4xx or 2xx accordingly.